### PR TITLE
Rename start and end datetime parameters; Refactor update dictionary creation method;

### PR DESCRIPTION
**Pull request by marjev**
2nd cleanup PR:

API endpoints should take start_datetime and end_datetime and add a 'name' key with title value in request data so that we can remove UPDATABLE_VOUCHER_FIELDS. We also don't need `retrieve_invoice_data` method since we can reuse `create_update_data_dict`.

@mjfrey, @vkaracic, @iivic can you guys please take a look at this?

This is a part of [SOL-1890](https://openedx.atlassian.net/browse/SOL-1890).

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @marjev! It looks like you're a member of a company that does contract work for edX. If you're doing this work as part of a paid contract with edX, you should talk to edX about who will review this pull request. If this work is not part of a paid contract with edX, then you should ensure that there is an OSPR issue to track this work in [JIRA](https://openedx.atlassian.net), so that we don't lose track of your pull request.

To automatically create an OSPR issue for this pull request, just visit this link: https://openedx-webhooks.herokuapp.com/github/process_pr?repo=edx%2Fecommerce&number=907

File: ecommerce/extensions/api/v2/tests/views/test_coupons.py [hunk 1]
```diff
@@ -394,7 +370,7 @@ def test_update_title(self):
         """Test updating a coupon's title."""
         data = {
             'id': self.coupon.id,
-            'title': 'New title'
+            'name': 'New title'
```
Comment (iivic) [inline ecommerce/extensions/api/v2/tests/views/test_coupons.py]: Why did you rename this field into name?

File: ecommerce/extensions/api/v2/views/coupons.py [hunk 2]
```diff
@@ -371,19 +340,24 @@ def update(self, request, *args, **kwargs):
...
-        if request_data_key in request_data:
-            value = request_data.get(request_data_key)
-            update_dict[update_dict_key] = prepare_course_seat_types(value) \
-                if update_dict_key == 'course_seat_types' else value
+        update_dict = {}
+
+        for field in fields:
+            if field in data:
+                value = prepare_course_seat_types(data.get(field)) if field == 'course_seat_types' else data.get(field)
+                update_dict[field.replace('invoice_', '')] = value
```
Comment (iivic) [inline ecommerce/extensions/api/v2/views/coupons.py:360]: Why is field.replace('invoice_', '') required here?

Comment (marjev) [reply ecommerce/extensions/api/v2/views/coupons.py:360]: Backbone uses `invoice_` prefix to mark the Invoice model fields. So, removing that prefix is necessary here to ensure that we can easily update associated Invoice object.

File: ecommerce/static/js/models/coupon_model.js [hunk 3]
```diff
@@ -237,11 +237,17 @@ define([
                     options.data = JSON.stringify(this.toJSON());
                 } else {
                     if (_.has(attributes, 'start_date')) {
-                        attributes.start_date = moment.utc(attributes.start_date);
+                        attributes.start_datetime = moment.utc(attributes.start_date);
+                        delete attributes.start_date;
```
Comment (vkaracic) [inline ecommerce/static/js/models/coupon_model.js]: Why is this being deleted?

File: ecommerce/extensions/api/v2/tests/views/test_coupons.py (hunk 1 above)
Comment (marjev) [reply ecommerce/extensions/api/v2/tests/views/test_coupons.py]: If the data contains `title`, it will only update title of the coupon product, since Product has title field. Since this is testing updating the voucher names, we have to provide name attribute for this to work most naturally. I will, however, update the docstring and name of the test method since it's not describing the real test scenario.

Comment (iivic) [reply ecommerce/extensions/api/v2/tests/views/test_coupons.py]: """Test updating a coupon's title.""" That's why it didn't make sense to me. Once you've done the update 👍

File: ecommerce/static/js/models/coupon_model.js (hunk 3 above)
Comment (marjev) [reply ecommerce/static/js/models/coupon_model.js]: I'll remove that line. TBH, I don't remember why I placed that there, since this was done so long ago.

**Comment by vkaracic** [comment]
:+1:
